pkg/ccl/backupccl/backup_planning.go

```diff
@@ -208,6 +208,84 @@ func spansForAllTableIndexes(
 	return spans
 }
 
+// getTableSpansToTimestampProtect returns non-overlapping spans of the tables
+// being backed up, for the protected timestamp record to cover. It handles
+// interleaved and non-interleaved tables differently as explained below, so as
+// to optimize for the number/size of spans that need to be timestamp protected.
+func getTableSpansToTimestampProtect(
```

Comment:
any reason we wouldn't replace `spans` with this for what we backup as well? might be good to get @pbardea's input on that, but I'm a little wary to backup one set of spans but protect another. Of course that is OK if what we're protecting is a more concise representation of the same thing, but if that is the case, then we can just use it all the time instead, and if it isn't, that seems likely to cause subtle problems.

Reply:
I did think about this, in my opinion, it should be fine. I just wasn't sure about making such an invasive change if we are considering backporting this?

Reply:
I'd lean towards not backporting, or waiting until 20.2.1 to do so.

Reply:
Yep, it should be fine to just use these merged spans as the spans of the backup job. (Which I think would be the preferred option, like David mentioned, to clearly keep the spans we protect in sync with the ones we restore.)

When we plan out the backup processors DistSQL's `PartitionSpans` will break up any larger spans into ranges.